### ISSUE TITLE
Provide a mechanism for users to say thanks or provide positive feedback on features.

### ISSUE THREAD

**Issue by NeilMacMullen** [microsoft/terminal#7428]
# Description of the new feature/enhancement

The terminal currently provides a "feedback"  menu item which links to the issue list here in GitHub.  This allows users to report problems or request features but it's less clear how they can show appreciation for existing features, indicate what is working well or just say "thanks" to the developers!   

# Proposed technical implementation details (optional)

I suggest adding something similar to the old Visual Studio/Office "send a smile"  options under the feedback menu.  This could link to single github issue to collect comments or could create a new issue with a special tag.

**Comment by NeilMacMullen** [comment]
Regardless of whether this goes anywhere, I just wanted to say thanks to the development team. When I first heard about a new terminal I couldn't really see the point but I'm now a convert.   Favourite features are multiple tabs and the ability to individually colour the tab headers.  Finally being able to use CTRL-C as "copy" in a powershell session is wonderful, even if it's taking me some time to train myself back into it! :-)  Also very much looking forward to seeing where the command-pallette takes us - I can see lots of useful stuff being layered on top of this.

**Comment by DHowett** [comment]
Thanks for the kind words :smile: The team always loves getting feedback, be it kudos or bug reports, so I very much appreciate your suggestion.

@cinnamon-msft Is there somewhere we can collect things like this? I don't necessarily want to make another place that people might file bugs (which we're prone to missing), but it might be nice to aggregate some of these things.

**Comment by cinnamon-msft** [comment]
I'm hesitant to add another item to the dropdown list inside the app. However, we could pin a "positivity" issue to the top of the issues list, so clicking the Feedback button would still bring you to it.

**Comment by WSLUser** [comment]
What about doing what the WSL team did and add a Discussion tab to the repo?

**Comment by alannt777** [comment]
GitHub Discussions is a beta feature. But I think the Terminal team can request early access.

**Comment by WSLUser** [comment]
Again WSL repo has it. So Terminal can too

**Comment by DHowett** [comment]
Well, since you mentioned WSL. I'm not sure that this is representative of the kind of content we'd want on our discussions page:

![image](https://user-images.githubusercontent.com/189190/98864185-5a726f80-241e-11eb-9db9-04271f3000ed.png)

That's just a separate bug tracker that looks like a forum but is still just as filled with bug reports as the _actual bug tracker_.

**Comment by WSLUser** [comment]
Yeah I noticed they migrated some issues to the discussions (looks like any issue that had been issued the discussion tag). I'm sure we can lay some ground rules and what should and shouldn't be placed in Discussions tab. One of the discussions can be dedicated to resolving the specific issue here.

**Comment by alannt777** [comment]
They are not really bugs but they are questions like in Stack Overflow

**Comment by Don-Vito** [comment]
I agree with Dustin that having an open discussion page might fork the issue tracker. 

Do we believe the tools for discussions and answering questions that we have today are not good enough for the target audience? I personally feel they are OK (after all I am managing a discussion with you right now). 

Going back to the original idea, as a new member of community I personally miss a way to express my opinion. And when I want to provide a feedback:
1. I don't really want to discuss with the entire community on how terribly wrong I am to not appreciate some default or animation. 
2. I also don't want to learn that there is a duplicate discussion from 5 years ago that is idle for the last year (that I wouldn't find even with Azure AI). 

When I have an idea and want to hear back I open an issue here.

So what I need is a way to provide the following one-way messages:
1. Feedback for features that exist
2. Voting for big changes that are planned to affect my experience by default, like default features, changes in look-n-feel, installation, etc. E.g., I wish we had a poll for MRU vs InOrder tabs to understand what should be a default for new installations.

**Comment by alannt777** [comment]
Look at most of issues with Issue-Question label, most of them are closed after they are answered. People who have the same question might not check the closed issues. But with Discussions, you can find both the questions and answers easily because their is no "closing" or "reopening". People just find it when they search for it.

**Comment by alannt777** [comment]
Look at most of issues with Issue-Question label, most of them are closed after they are answered. People who have the same question might not check the closed issues. But with Discussions, you can find both the questions and answers easily because their is no "closing" or "reopening". People just find it when they search for it.

**Comment by Don-Vito** [comment]
> 
> 
> Look at most of issues with Issue-Question label, most of them are closed after they are answered. People who have the same question might not check the closed issues. But with Discussions, you can find both the questions and answers easily because their is no "closing" or "reopening". People just find it when they search for it.

IMHO this approach will work as long as you have several dozens of such discussions. Afterwards we will have pages and we will still need to do the same search we have today.. unless I am missing something :blush:

**Comment by alannt777** [comment]
> > Look at most of issues with Issue-Question label, most of them are closed after they are answered. People who have the same question might not check the closed issues. But with Discussions, you can find both the questions and answers easily because their is no "closing" or "reopening". People just find it when they search for it.
> 
> IMHO this approach will work as long as you have several dozens of such discussions. Afterwards we will have pages and we will still need to do the same search we have today.. unless I am missing something 😊

Why do you think Stack Overflow is better than Issues for asking questions/finding answers? Same thing essentially 😊

**Comment by Don-Vito** [comment]
> 
> 
> > > Look at most of issues with Issue-Question label, most of them are closed after they are answered. People who have the same question might not check the closed issues. But with Discussions, you can find both the questions and answers easily because their is no "closing" or "reopening". People just find it when they search for it.
> > 
> > 
> > IMHO this approach will work as long as you have several dozens of such discussions. Afterwards we will have pages and we will still need to do the same search we have today.. unless I am missing something 😊
> 
> Why do you think Stack Overflow is better than Issues for asking questions/finding answers? Same thing essentially 😊

1. Mostly because Google search provides you a preview 😝 
2. In many cases there is not another choice as product/tracker is closed source / the community is closed or "hating"

But yeah, in many cases it works and better searchable. However the question is does it justify the a risk for splitting the tracker?
And if we do feel strong about it, than probably it should be StackOverflow :smile:

**Comment by alannt777** [comment]
> In many cases there is not another choice as product / tracker / community are closed (or "hating")

You *can* restrict commenting to collaborators as you could do with Issues

> However the question is does it justify the a risk for splitting the tracker?

It is *not* a bug tracker

>  probably it should be StackOverflow 😄

I don't mind it being Stack Overflow or Reddit or whatever

**Comment by Don-Vito** [comment]
> 
> You _can_ restrict commenting to collaborators as you could do with Issues
> 

What I meant is that sometimes people ask in StackOverflow because they don't feel comfortable to ask the community.. that in some projects is not really nice with "noobs"

**Comment by alannt777** [comment]
> > You _can_ restrict commenting to collaborators as you could do with Issues
> 
> What I meant is that sometimes people ask in StackOverflow because they don't feel comfortable to ask the community.. that in some projects is not really nice with "noobs"

Valid concern. But let's see what the core team thinks about our discussions so far.

**Comment by itsme-alan** [comment]
UPDATE: GitHub Discussions can now be enabled in the Repository Settings

@Don-Vito No one from the team seems to be interested in our conversation 😄

**Comment by zadjii-msft** [comment]
I'm sorry, I just don't think I see the value in Discussions. To me, it looks like another fragmented place for people to ask duplicate questions. I honestly really doubt that the people who can't search the _one giant list of everything_ will be able to figure out searching in _both_ locations. It raises questions like "what's an issue vs discussion?". It creates another list of things that we'll have to ensure to triage regularly, instead of having all the answers in one place. 

Maybe a discussion _is_ an issue. Then we need to convert the discussion back to an issue - how does _that_ work? How are the comments maintained?

It also creates two different UX's for communicating on GH - right now in an issue, everything is in one list. With discussions, there's top level comments, and those can have replies. I guarantee you that people will reply to other top-level comments _in their own top-level comment_. 

IDK, this just doesn't seem valuable to me.

**Comment by itsme-alan** [comment]
@zadjii-msft If you feel that way, "we can consider other alternative solutions for *this* issue

**Comment by NeilMacMullen** [comment]
This topic seems to have drifted quite a bit.  The _original_ suggestion was simply to provide an low-friction way for users to express  +positive+ feedback.  It's clearly inappropriate to raise a github issue for that purpose since those tend to be associated with problems or change-requests.  FWIW I don't think there's a lot of added value in allowing discussion around positive comments either - I just thought it would be nice for the development team to see that people appreciated the product.

**Comment by NeilMacMullen** [comment]
Let me put it this way ... :-)  

"_As a user_ of the terminal I would like to be able to quickly offer positive comments about  the product instead of being forced to raise a scary looking **bug-report** when I click on the Feedback menu item"

**Comment by itsme-alan** [comment]
Do you remember the issue options config PR that I did. We can add an option like "General Feedback" that links to an issue thread like a mentioned before.

When they click on feedback, they will be dropped into https://github.com/microsoft/terminal/issues/new/choose where they will find this option

**Comment by itsme-alan** [comment]
Hey wait! Isn't Feedback Hub the best place for the original ask?

**Comment by zadjii-msft** [comment]
Feedback hub isn't even the best way to report _negative feedback_ 🤣. Maybe I'm just being daft, but it certainly doesn't look like there's a way to use the Feedback hub for positive feedback, only "report a problem" and "suggest a feature"

**Comment by itsme-alan** [comment]
@zadjii-msft But I have seen people give positive feedback as well

**Comment by NeilMacMullen** [comment]
>Feedback hub isn't even the best way to report negative feedback

indeed.. "where feedback goes to die" !  Maybe even just a link to "star" the terminal repo would be a start?
